smart-contract/assembly/contracts/internals/chunks.ts

```diff
@@ -1,120 +1,124 @@
- * @returns The chunk data.
+ * @returns The chunk data as a StaticArray<u8>.
  * @throws If the chunk is not found in storage.
  */
 export function _getFileChunk(
-  filePathHash: StaticArray<u8>,
+  hashLocation: StaticArray<u8>,
   index: u32,
 ): StaticArray<u8> {
-  assert(Storage.has(_getChunkKey(filePathHash, index)), 'Chunk not found');
-  return Storage.get(_getChunkKey(filePathHash, index));
+  assert(Storage.has(fileChunkKey(hashLocation, index)), 'Chunk not found');
+  return Storage.get(fileChunkKey(hashLocation, index));
 }
 
 /**
  * Gets the total number of chunks for a file.
- * @param filePathHash - The hash of the file path.
+ * @param hashLocation - The hash of the file location.
  * @returns The total number of chunks, or 0 if not set.
  */
-export function _getTotalChunk(filePathHash: StaticArray<u8>): u32 {
-  if (!Storage.has(_getTotalChunkKey(filePathHash))) return 0;
-  return bytesToU32(Storage.get(_getTotalChunkKey(filePathHash)));
+export function _getTotalChunk(hashLocation: StaticArray<u8>): u32 {
+  if (!Storage.has(fileChunkCountKey(hashLocation))) return 0;
+  return bytesToU32(Storage.get(fileChunkCountKey(hashLocation)));
 }
 
+/* -------------------------------------------------------------------------- */
+/*                                   DELETE                                   */
+/* -------------------------------------------------------------------------- */
+
 /**
- * Generates the storage key for the number of chunks of a file.
- * @param filePathHash - The hash of the file path.
- * @returns The storage key for the number of chunks.
+ * Removes a range of chunks for a given file from storage.
+ * @param hashLocation - The hash of the file location.
+ * @param start - The starting index of the range to remove (inclusive).
+ * @param end - The ending index of the range to remove (inclusive).
  */
-export function _getTotalChunkKey(
-  filePathHash: StaticArray<u8>,
-): StaticArray<u8> {
-  return CHUNK_NB_TAG.concat(filePathHash);
+export function _removeChunksRange(
+  hashLocation: StaticArray<u8>,
+  start: u32 = 0,
+  end: u32 = 0,
+): void {
+  for (let i = u32(start); i <= end; i++) {
+    Storage.del(fileChunkKey(hashLocation, i));
+  }
 }
 
 /**
- * Generates the storage key for a specific chunk of a file.
- * @param filePathHash - The hash of the file path.
- * @param index - The index of the chunk.
- * @returns The storage key for the chunk.
+ * Deletes all chunks and metadata of a given file from storage.
+ * @param hashLocation - The hash of the file location.
  */
-export function _getChunkKey(
-  filePathHash: StaticArray<u8>,
-  index: u32,
-): StaticArray<u8> {
-  return FILE_TAG.concat(filePathHash)
-    .concat(CHUNK_TAG)
-    .concat(u32ToBytes(index));
-}
+export function _deleteFile(hashLocation: StaticArray<u8>): void {
+  // Remove all chunks
+  const chunkKeys = Storage.getKeys(
+    FILE_TAG.concat(hashLocation).concat(CHUNK_TAG),
```

Comment:
wouldn't it be simpler to just delete everything with the prefix `FILE_TAG.concat(hashLocation)` ? it would remove chunks, chunk counts, metadata and anything file-specific that we might want to add in the future so we don't forget to delete it here